tools/cli/adminDBScanCommand.go

```diff
@@ -350,3 +369,32 @@ func getSQLParser(encodingType common.EncodingType, decodingTypes ...common.Enco
 	}
 	return parser
 }
+
+func initializeCassandraShardStore(
+	c *cli.Context,
+	currentClusterName string,
+	logger log.Logger,
+) persistence.ShardStore {
+	client, session := connectToCassandra(c)
+	shardStore := cassandra.NewShardPersistenceFromSession(client, session, currentClusterName, loggerimpl.NewNopLogger())
+	return shardStore
+}
+
+func initializeSQLShardStore(
+	c *cli.Context,
+	currentClusterName string,
+	logger log.Logger,
+) persistence.ShardStore {
+	sqlDB := connectToSQL(c)
+	encodingType := c.String(FlagEncodingType)
+	decodingTypesStr := c.StringSlice(FlagDecodingTypes)
+	var decodingTypes []common.EncodingType
+	for _, dt := range decodingTypesStr {
+		decodingTypes = append(decodingTypes, common.EncodingType(dt))
+	}
+	shardStore, err := sql.NewShardPersistence(sqlDB, currentClusterName, logger, getSQLParser(common.EncodingType(encodingType), decodingTypes...))
+	if err != nil {
+		ErrorAndExit("Failed to get shard store from sql config", err)
+	}
+	return shardStore
```

Comment:
Maybe return err also? Exit paths could be stored in a single place

Reply:
It's to keep the function signature the same as others.